### PR TITLE
Feature/gh 13428 message component

### LINKED ISSUE #13428: Reusable Message Component with body
New message component with additional body content. For now it will be only additional component, but we will use it in 5.0 to refactor GlobalMessageComponent.

### PR DESCRIPTION
**Pull request by dydome**
close #13428

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/misc/message/message.component.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  type: GlobalMessageType;
+
+  @Output()
+  clear: EventEmitter<void> = new EventEmitter();
+
+  iconTypes = ICON_TYPE;
+
+  constructor() {}
+
+  get getCssClassesForMessage(): any {
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.ts]: Change it from any to Record<string, boolean>

File: projects/storefrontlib/src/cms-components/misc/message/message.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return {
+      'message-success': this.type === GlobalMessageType.MSG_TYPE_CONFIRMATION,
+      'message-info': this.type === GlobalMessageType.MSG_TYPE_INFO,
+      'message-warning': this.type === GlobalMessageType.MSG_TYPE_WARNING,
+      'message-danger': this.type === GlobalMessageType.MSG_TYPE_ERROR,
+    };
+  }
+
+  get getIconType(): any {
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.ts]: Also use the correct type here (ICON_TYPE)

File: projects/storefrontlib/src/cms-components/misc/message/message.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      }).compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MessageComponent);
+    messageComponent = fixture.componentInstance;
+  });
+
+  it('should create message component', () => {
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.spec.ts:41]: In unit tests check things like type, ng-content, title and clear

File: projects/storefrontlib/src/cms-components/misc/message/message.component.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+})
+export class MessageComponent {
+  @Input()
+  text: string;
+
+  @Input()
+  type: GlobalMessageType;
+
+  @Output()
+  clear: EventEmitter<void> = new EventEmitter();
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.ts]: I wonder if that should be clear or close

Comment (dydome) [reply projects/storefrontlib/src/cms-components/misc/message/message.component.ts]: close is better

File: projects/storefrontlib/src/cms-components/misc/message/message.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    fixture.detectChanges();
+
+    expect(component.getIconType).toEqual(ICON_TYPE.SUCCESS);
+  });
+
+  it('should show <ng-content> content', () => {
+    const testFixture = TestBed.createComponent(TestHostComponent);
+    const element = testFixture.debugElement.query(By.css('cx-message'))
+      .nativeElement;
+    console.log('****1 element', element);
```
Comment (dydome) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.spec.ts]:
```suggestion
```

**Review by marlass — DISMISSED** [review]

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

File: projects/storefrontlib/src/cms-components/misc/message/message.component.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<div [ngClass]="getCssClassesForMessage" class="message">
```
Comment (morganm58) [inline projects/storefrontlib/src/cms-components/misc/message/message.component.html:1]: The classes are missing the cx-name rule. For example message should be cx-message

File: projects/storefrontstyles/scss/cxbase/blocks/message.scss [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+  &-content {
+    display: flex;
+    flex-direction: column;
+    margin: auto;
+    max-width: var(--cx-page-width-max);
+  }
+
+  &-header {
+    @include type('6', 'semi');
```
Comment (morganm58) [inline projects/storefrontstyles/scss/cxbase/blocks/message.scss:17]: the font-weight should be included separately.
```
@include type('6');
font-weight: var(--cx-font-weight-semi); 
```

File: projects/storefrontstyles/scss/cxbase/blocks/message.scss [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+      font-weight: $font-weight-bold;
+      height: var(--cx-height, 26px);
+      position: var(--cx-position, relative);
+      text-align: var(--cx-text-align, center);
+      width: var(--cx-width, 26px);
+    }
+  }
+
+  &-success {
+    background-color: #f0fbe4;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/cxbase/blocks/message.scss:58]: we should be using color variable for all colors

File: projects/storefrontstyles/scss/cxbase/blocks/message.scss [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+    .message-icon {
+      cx-icon {
+        color: #fbba22;
+      }
+    }
+  }
+
+  .close {
+    font-weight: var(--cx-font-weight, var(--cx-font-weight-normal));
```
Comment (morganm58) [inline projects/storefrontstyles/scss/cxbase/blocks/message.scss:102]: it should be `font-weight: var(--cx-font-weight-normal);`

File: projects/storefrontstyles/scss/cxbase/blocks/message.scss [hunk 10]
```diff
@@ -0,0 +1,107 @@
+@import '../../theme';
+@import '../../mixins';
+
+.message {
+  color: var(--cx-color, var(--cx-color-text));
```
Comment (morganm58) [inline projects/storefrontstyles/scss/cxbase/blocks/message.scss:5]: should be `color: var(--cx-color-text);`

**Pull request merged**